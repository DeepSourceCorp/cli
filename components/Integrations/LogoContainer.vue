<template>
  <div class="p-1 rounded-sm bg-ink-300">
    <img :src="logo" alt="Integration logo" :class="['flex-shrink-0', dimensions]" />
  </div>
</template>

<script lang="ts">
import { ZIcon } from '@deepsource/zeal'
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component({
  components: {
    ZIcon
  }
})

/**
 * Component that is responsible to render integration logo container on detailed view
 */
export default class LogoContainer extends Vue {
  @Prop({ required: true })
  logo: string

  @Prop({ default: 'w-5 h-5' })
  dimensions: string
}
</script>
