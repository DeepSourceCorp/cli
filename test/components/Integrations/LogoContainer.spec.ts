import '@testing-library/jest-dom'
import { render } from '@testing-library/vue'

import LogoContainer from '~/components/Integrations/LogoContainer.vue'

describe('[[ LogoContainer ]]', () => {
  test('renders the integration logo in a container', () => {
    const { html } = render(LogoContainer, {
      props: {
        logo: 'https://static.deepsource.io/integration_logos/slack.svg'
      },
      stubs: {
        ZIcon: true
      }
    })

    expect(html()).toMatchSnapshot()
  })
})
