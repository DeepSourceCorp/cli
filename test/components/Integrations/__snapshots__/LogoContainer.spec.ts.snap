// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ LogoContainer ]] renders the integration logo in a container 1`] = `"<div class=\\"p-1 rounded-sm bg-ink-300\\"><img src=\\"https://static.deepsource.io/integration_logos/slack.svg\\" alt=\\"Integration logo\\" class=\\"flex-shrink-0 w-5 h-5\\"></div>"`;
